File: Lecture 3/practice/ListNPrimes.java
public class ListNPrimes {
    
    public static boolean isPrime(int num){
        if (num < 2) 
            return false;
        for(int i = 2; i * i <= num; i++){
            if(num % i == 0){
                return false;
            }
        }
        return true;
    }

    public static void listPrimes(int num, int prime, int count){
        if(count == num){
            return;
        }

        if(isPrime(prime)){
            System.out.print(prime + " ");
            count++;
        }

        listPrimes(num, prime + 1, count);
    }

    public static void main(String[] args) {
        int num = 5;
        System.out.println("First " + num + " primes are: ");
        listPrimes(num, 2, 0);
    }
}

File: Lecture 3/practice/ListNPrimes.py
# 3. Liệt kê n số nguyên tố đầu tiên
import math


def isPrime(num):
    if num < 2:
        return False
    for i in range(2, int(math.sqrt(num) + 1)):
        if num % i == 0:
            return False
    return True


def listNPrimes(num, firstPrime=2, prime=[]):
    if len(prime) == num:
        return prime
    else:
        if isPrime(firstPrime):
            prime.append(firstPrime)
        return listNPrimes(num, firstPrime + 1, prime)


num = int(input("Upperbound:"))
print(listNPrimes(num))
